presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -174,7 +174,12 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableSet.of(new SimplifyCountOverConstant())),
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
                 new MergeWindows(),
-                new ReorderWindows(), // Should be after MergeWindows to avoid unnecessary reordering of mergeable windows
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableSet.of(
```

Comment:
You should add ReorderWindows to the "legacy rules" argument to IterativeOptimizer. That way, it's possible to turn on the old behavior if there are any issues. Once it's been running fine for a couple of releases, we can just get rid of it.